File: MVVMDemo/app/src/main/java/com/example/a14512/mvvmdemo/net/ApiService.java
package com.example.a14512.mvvmdemo.net;

import com.example.a14512.mvvmdemo.mode.News;

import retrofit2.http.POST;
import retrofit2.http.Query;
import rx.Observable;

/**
 * @author 14512 on 2018/8/20
 */
public interface ApiService {
    String BASE_URL = "http://api.tianapi.com/";

    /**
     * get news
     * @param key
     * @param num
     * @return
     */
    @POST("it")
    Observable<News> getNews(@Query("key")String key, @Query("num")String num);
}

File: GithubDemo/app/src/main/java/com/example/githubdemo/net/ApiService.kt
package com.example.githubdemo.net

import com.example.githubdemo.moudels.User
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * @author 14512 on 2019/4/16
 */
interface ApiService {
    @GET("users/airbnb/repos")
    fun getUsers(@Query("page") page: Int): Observable<ApiResult<List<User>>>
}
